.github/workflows/deploy.yml

```diff
@@ -0,0 +1,86 @@
+
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        description: 'The terraform workspace to target'
+        required: true
+        type: string
+      version_tag:
+        description: 'Container version to deploy'
+        required: true
+        type: string
+    outputs:
+      cluster_config:
+        description: 'JSON string of cluster configuration'
+        value: ${{ jobs.terraform_workflow_job.outputs.cluster_config_json }}    
+
+jobs:
+  terraform_workflow_job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Extract branch name
+        run: |
+          if [ "${{ github.head_ref }}" == "" ]; then
+            echo BRANCH_NAME=main >> $GITHUB_ENV
+          else
+            echo BRANCH_NAME=${{ github.head_ref }} >> $GITHUB_ENV
+          fi
+        id: extract_branch
+      - uses: unfor19/install-aws-cli-action@v1
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.2.3
+          terraform_wrapper: false
+      - name: Configure AWS Credentials For Terraform
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-1
+          role-duration-seconds: 3600
+          role-session-name: GithubActions
+      - uses: webfactory/ssh-agent@v0.5.4
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY_ALLOW_LIST_REPOSITORY }}
+
+      - name: Lint Terraform
+        id: tf_lint
+        run: terraform fmt -check -recursive
+        working-directory: ./terraform
+        continue-on-error: true
```

Comment:
Looking at the main flow this workflow runs after a few other steps...would we want this as early in the overall process as possible so we can fail on it to stop badly formatted terraform being committed to main? Like some kind of lint step (that could eventually include PHP, Go and JS linting) being one of the first?

Reply:
This runs in the PR workflow as well so my thinking was that it should get caught then as part of a review. The combined lint step is a good idea, we could do it as part of another ticket?

Reply:
Ah yeah good point. But yeah an extra ticket for lining as a first step in a new ticket sounds good - I'll drop one in now.